### DISCUSSION TITLE
Beelink Mini S12 Pro + Ubuntu 24.04 LTS

### DISCUSSION THREAD

**Discussion by nehalvpatel** [OpenSurv/OpenSurv#23] [Hardware + Operating system]
### Specs
- Intel Alder Lake (12th Gen) N100 4C/4T (up to 3.4GHz)
- 16GB DDR4 (3200MHz)
- Intel UHD Graphics 24EUs
- Dual HDMI2.0 4K 60Hz

### Setup
- OpenSurv running in an Ubuntu 24.04 VM within Proxmox.
- Integrated GPU fully passed through to the VM using [this guide](https://3os.org/infrastructure/proxmox/gpu-passthrough/igpu-passthrough-to-vm/#windows-virtual-machine-igpu-passthrough-configuration).
- 33 RTSP streams from Unifi Protect with 6 columns, always visible, just 1 monitor.

### Results

So good! It takes a little bit to start as expected, but it has been really solid. I noticed some micro skips in the feeds so I upped the core count from 2 to 4 and the RAM from 4GB to 6GB.

I still see some micro stutters, so I'm thinking it's a GPU thing? Would love any feedback on this.

Overall, this project is a dream. THANK YOU!

The stupid little Unifi Viewport costs as much as it does and only does a 4x4 grid.

### Screenshots

<details>

<summary><strong>📸 Proxmox setup</strong></summary>

![CleanShot 2024-12-14 at 07 57 56@2x](https://github.com/user-attachments/assets/a49338e7-f2bc-4bfd-96d5-f4a2e4a443e9)

</details>

<details>

<summary><strong>📸 Resource usage</strong></summary>

![CleanShot 2024-12-14 at 08 02 03@2x](https://github.com/user-attachments/assets/19041080-9484-449a-a755-9db290b795ed)

</details>

**Comment by SvenVD** [comment]
Thanks for the extensive reporting back! 33 streams on a VM is pretty cool. You can see the load of the gpu with ` i7s` and `gputop` commands, I'm curious to see if the GPU is maxed out and if that is causing the stutters. Otherwise you can reduce the stream count as a test and see if stuttering is gone. It could be a network bottleneck also.

**Reply by nehalvpatel** [reply]
I used `intel_gpu_top` and it looks like the Render/3D engine usage ranges from 90-97% consistently. So it looks like it’s the GPU, then? I checked the network Rx rate and it was something like 12 megabits/s, haha.

I didn’t use any flags in my config, do you think anything would help?

![IMG_0148](https://github.com/user-attachments/assets/5b279b63-99f5-4926-b4e7-ca740544848f)

**Comment by nehalvpatel** [comment]
I might be seeing some memory leak issue, but not sure.

![IMG_7686](https://github.com/user-attachments/assets/a0aa2375-a108-473f-8efd-e106eedf2b3c)

**Reply by SvenVD** [reply]
Do you see any cumulating processes or errors in the logs:/home/opensurv/logs/main.log.  Depending on what your graph is showing, it could be looking at the wrong memory statistic you should be looking at  ` cat /proc/meminfo | grep -i available`  is that also decreasing?

**Reply by nehalvpatel** [reply]
I can take a closer look later today, but actually the reason I started looking into this was that I noticed the live streams were drifting behind several seconds from real time.

To try to fix it, I updated the global `/home/opensurv/.config/mpv/mpv.conf` to include this:
```
untimed=yes
profile=low-latency
```

But I am still seeing the drift issue.

I also noticed the PIDs for `mpv` are also ballooning into the hundreds of thousands, maybe it keeps getting relaunched?

I'm gonna try to enable logging and see what's really going on.

**Reply by SvenVD** [reply]
I think too much streams causing all kind of timeouts and loss of tracking of the windows, causing cumulation. But the logs will tell you that.

**Comment by sborn** [comment]
Sorry to sound pedestrian, but why did you run in a Proxmox VM rather than installing Ubuntu directly?

If I bought one and installed 24.04 directly, would I have the same performance?

**Reply by sborn** [reply]
I can confirm that it works for at least 9 streams on an S12 Pro running 24.04.3 bare metal
